accounts/serializers.py

```diff
@@ -25,12 +30,38 @@ class Meta:
             "user_permissions",
         )
 
+        read_only_fields = (
+            "pennid",
+            "last_name",
+            "username",
+            "email",
+            "groups",
+            "product_permission",
+            "user_permissions",
+        )
+
+    # Users are pulled from Penn DB, so come with no preferred
+    # name. Thus, this logic only needs to happen on update.
+    def update(self, instance, validated_data):
+        if validated_data["get_preferred_name"]:
```

Comment:
this should be `if "get_preferred_name" in validated_data`

Reply:
Also, I'm not too happy with how the logic is structured here. I know we came up with it together, but it feels like there should be a slightly cleaner way of doing things.